test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Smart/SmartSearchTests.cs

```diff
@@ -0,0 +1,358 @@
+                ModelInfoProvider.Instance.Version == FhirSpecification.R4B)
+            {
+                _dataStoreSearchParameterValidator.ValidateSearchParameter(default, out Arg.Any<string>()).ReturnsForAnyArgs(x =>
+                {
+                    x[1] = null;
+                    return true;
+                });
+
+                _searchParameterSupportResolver.IsSearchParameterSupported(Arg.Any<SearchParameterInfo>()).Returns((true, false));
+
+                _contextAccessor = _fixture.FhirRequestContextAccessor;
+
+                _fhirOperationDataStore = _fixture.OperationDataStore;
+                _fhirStorageTestHelper = _fixture.TestHelper;
+                _scopedDataStore = _fixture.DataStore.CreateMockScope();
+
+                _searchParameterDefinitionManager = _fixture.SearchParameterDefinitionManager;
+                _supportedSearchParameterDefinitionManager = _fixture.SupportedSearchParameterDefinitionManager;
+
+                _typedElementToSearchValueConverterManager = await CreateFhirTypedElementToSearchValueConverterManagerAsync();
+
+                _searchIndexer = new TypedElementSearchIndexer(
+                    _supportedSearchParameterDefinitionManager,
+                    _typedElementToSearchValueConverterManager,
+                    Substitute.For<IReferenceToElementResolver>(),
+                    ModelInfoProvider.Instance,
+                    NullLogger<TypedElementSearchIndexer>.Instance);
+
+                ResourceWrapperFactory wrapperFactory = Mock.TypeWithArguments<ResourceWrapperFactory>(
+                    new RawResourceFactory(new FhirJsonSerializer()),
+                    new FhirRequestContextAccessor(),
+                    _searchIndexer,
+                    _searchParameterDefinitionManager,
+                    Deserializers.ResourceDeserializer);
+
+                _searchParameterStatusManager = _fixture.SearchParameterStatusManager;
+
+                _searchService = _fixture.SearchService.CreateMockScope();
+
+                _contextAccessor = _fixture.FhirRequestContextAccessor;
+
+                var smartBundle = Samples.GetJsonSample<Bundle>("SmartPatientA");
+                foreach (var entry in smartBundle.Entry)
+                {
+                    await PutResource(entry.Resource);
+                }
+            }
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        [SkippableFact]
+        public async Task GivenScopesWithReadForAllResources_WhenRevIncludeObservations_PatientAndObservationReturned()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            // try to query both the patient and the observation using revinclude
```

Comment:
Nit; `Patient resource` and `Observation resource` for clarity. Same goes for other comments in this file.

Reply:
Updated.